r/tests/testthat/test-dplyr-join.R

```diff
@@ -371,3 +371,24 @@ test_that("full joins handle keep", {
     )
   }
 })
+
+test_that("right_join correctly coalesces keys", {
```

Comment:
Instead of another test, could we modify the existing `left` and `to_join` to have keys that don't fully overlap?

https://github.com/apache/arrow/blob/785ab5f3c3aa4e5f2e30655d4765ca729bee7aa2/r/tests/testthat/test-dplyr-join.R#L20-L27

Then we get coverage from:

https://github.com/apache/arrow/blob/785ab5f3c3aa4e5f2e30655d4765ca729bee7aa2/r/tests/testthat/test-dplyr-join.R#L143-L152

Reply:
Good call, done now.